### PR TITLE
Avatar Thumbnail Fix

### PR DESCRIPTION
**Pull request by denjello**
This is a fix from #1850 for showing the appropriate avatar image instead of mirroring the active one.

### CODE REVIEW

**Comment by denjello** [comment]
Not sure what borked exactly…

File: app/models/user.rb [hunk 1]
```diff
@@ -284,7 +284,7 @@ def inactive_message
     I18n.t(:inactive_html, path_to_contact: '/contact').html_safe
   end
 
-  def avatar_url(size=nil)
+  def avatar_url(size=nil, avatar_kind)
```
Comment (gdpelican) [inline app/models/user.rb]: Cool, so you should be able to fix your error by making this `avatar_kind=nil`

Reason being we call this method elsewhere with no arguments, so the method needs to function without having anything passed in.

Other than that this fix looks like it should work. :tropical_fish:

**Comment by gdpelican** [comment]
Hey @denjello, would you mind having a poke through https://github.com/loomio/loomio/wiki/How-to-squash, which will allow us to accept this PR? We generally don't accept pull requests with multiple commits, because they muddy up the commit history on master.

Happy to help with other issues if they arise. :8ball:

**Comment by denjello** [comment]
I looked at the spec of `spec/models/user_spec.rb` again, and it really seems strange and potentially entirely wrong. I can't figure out why if the avatar_kind is 'uploaded' it would ever return a gravatar address...

```
describe "avatar_url" do
    it "returns gravatar url if avatar_kind is 'gravatar'" do
      user.should_receive(:gravatar_url).and_return('www.gravatar/spike')
      user.avatar_kind = 'gravatar'
      user.avatar_url(:small).should == 'www.gravatar/spike'
    end

    context "where avatar_kind is 'uploaded'" do
      before do
        @uploaded_avatar = double "paperclip_image"
        user.should_receive(:uploaded_avatar).and_return(@uploaded_avatar)
      end
      it "returns medium url if no size is specified" do
        @uploaded_avatar.should_receive(:url).with(:medium).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:medium).should == 'www.gravatar/uploaded/mike'
      end
      it "returns large url if large size is specified" do
        @uploaded_avatar.should_receive(:url).with(:large).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:large).should == 'www.gravatar/uploaded/mike'
      end
      it "returns medium url if medium size is specified" do
        @uploaded_avatar.should_receive(:url).with(:medium).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:medium).should == 'www.gravatar/uploaded/mike'
      end
      it "returns small url if small size is specified" do
        @uploaded_avatar.should_receive(:url).with(:small).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:small).should == 'www.gravatar/uploaded/mike'
      end
    end
  end
```

**Comment by denjello** [comment]
The spec needs to be changed to reflect the fix because user.avatar_kind is not being passed to the avatar_url function. 

e.g.
```
  describe "avatar_url" do
    it "returns gravatar url if avatar_kind is 'gravatar'" do
      user.should_receive(:gravatar_url).and_return('www.gravatar/spike')
      user.avatar_kind = 'gravatar'
      user.avatar_url(:small,user.avatar_kind).should == 'www.gravatar/spike'
    end

    context "where avatar_kind is 'uploaded'" do
      before do
        @uploaded_avatar = double "paperclip_image"
        user.should_receive(:uploaded_avatar).and_return(@uploaded_avatar)
      end
      it "returns medium url if no size is specified" do
        @uploaded_avatar.should_receive(:url).with(:medium).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:medium,user.avatar_kind).should == 'www.gravatar/uploaded/mike'
      end
      it "returns large url if large size is specified" do
        @uploaded_avatar.should_receive(:url).with(:large).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:large,user.avatar_kind).should == 'www.gravatar/uploaded/mike'
      end
      it "returns medium url if medium size is specified" do
        @uploaded_avatar.should_receive(:url).with(:medium).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:medium,user.avatar_kind).should == 'www.gravatar/uploaded/mike'
      end
      it "returns small url if small size is specified" do
        @uploaded_avatar.should_receive(:url).with(:small).and_return('www.gravatar/uploaded/mike')
        user.avatar_kind = 'uploaded'
        user.avatar_url(:small,user.avatar_kind).should == 'www.gravatar/uploaded/mike'
      end
    end
  end
```

**Comment by denjello** [comment]
forgot one of the `@uploaded_avatar.should_receive` corrections.
This should pass.

**Pull request closed without merging**